src/window.rs

```diff
@@ -363,14 +363,100 @@ fn stringify_message_content(message: &TelegramMessage, chat: &Chat) -> String {
+                    )
+                }
+            }
+        }
+        MessageContent::MessagePinMessage(data) => {
+            let message_id = data.message_id;
+            let message_ = chat.history().message_by_id(message_id);
+            let short_message = match message_ {
+                Some(data) => match data.content().0 {
+                    MessageContent::MessageText(data) => {
+                        let msg = data.text.text;
+                        if msg.chars().count() > 17 {
+                            gettext!("«{}»", msg.chars().take(16).collect::<String>())
+                        } else {
+                            gettext!("«{}»", &msg)
+                        }
+                    }
+                    MessageContent::MessagePhoto(_) => gettext("a photo"),
+                    MessageContent::MessageVideo(_) => gettext("a video"),
+                    MessageContent::MessageSticker(data) => {
+                        gettext!("a {} sticker", data.sticker.emoji)
+                    }
+                    MessageContent::MessageAnimation(_) => gettext("a GIF"),
+                    MessageContent::MessageDocument(_) => gettext("a file"),
+                    MessageContent::MessageAudio(_) => gettext("an audio file"),
+                    _ => gettext("a message"),
+                },
+                None => gettext("a deleted Message"),
+            };
+            gettext!(
+                "{} pinned {}",
+                sender_name(&message.sender_id, chat),
+                short_message
+            )
+        }
+        MessageContent::MessageChatChangeTitle(data) => {
+            let title = &data.title;
+            let type_ = chat.type_();
+            match type_ {
+                ChatType::Supergroup(data) => {
+                    let is_channel = data.is_channel();
+                    if is_channel {
+                        gettext!("Channel name was changed to «{}»", title)
+                    } else {
+                        gettext!(
+                            "{} changed group name to «{}»",
+                            sender_name(&message.sender_id, chat),
+                            title
+                        )
+                    }
+                }
+                _ => {
+                    gettext!(
+                        "{} changed group name to «{}»",
+                        sender_name(&message.sender_id, chat),
+                        title
+                    )
+                }
+            }
+        }
+        MessageContent::MessageChatJoinByLink => {
+            gettext!(
+                "{} joined the group via invite link",
+                sender_name(&message.sender_id, chat)
+            )
+        }
+        MessageContent::MessageChatJoinByRequest => {
+            gettext!(
+                "{} joined the group via invite link",
+                sender_name(&message.sender_id, chat)
+            )
+        }
```

Comment:
Why don't you join these 2 match arms with an `|` as they seem to do the same thing?